### PR TITLE
drivers: stm32_iwdg: fix config + prepare for watchdog early interrupt

### PR DESCRIPTION
**Pull request by etienne-lms**
Fix configuration of the watchdog + enable service to get STM32 IWDG timeout range + prepare for watch early interrupt.

### CODE REVIEW

**Review by jforissier — COMMENTED** [review]
One question on "drivers: stm32_iwdg: provide timeout range".
For all other commits:
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/drivers/stm32_iwdg.c [hunk 1]
```diff
@@ -188,6 +188,20 @@ static struct stm32_iwdg_device *wdt_chip_to_iwdg(struct wdt_chip *chip)
...
 }
 
+static TEE_Result iwdg_wdt_init(struct wdt_chip *chip,
+				unsigned long *min_timeout,
+				unsigned long *max_timeout)
+{
+	struct stm32_iwdg_device *iwdg = wdt_chip_to_iwdg(chip);
+
+	/* Be safe and expect any counter to be above 2 */
+	*min_timeout = 3 * IWDG_PRESCALER_256 / clk_get_rate(iwdg->clk_lsi);
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/drivers/stm32_iwdg.c:193]: Can't be negative but can be null.
I'll add a test.

**Comment by jforissier** [comment]
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by GseoC** [comment]
`Reviewed-by: Gatien Chevallier <gatien.chevallier@foss.st.com>`

**Comment by etienne-lms** [comment]
Review tags applied.

**Comment by jforissier** [comment]
@etienne-lms in commit "drivers: stm32_iwdg: enable bus clock once for all", your S-o-b should come first since you are the author.

**Comment by etienne-lms** [comment]
> @etienne-lms in commit "drivers: stm32_iwdg: enable bus clock once for all", your S-o-b should come first since you are the author.

I've updated messages of commits "drivers: stm32_iwdg: provide timeout range" and
"drivers: stm32_iwdg: enable bus clock once for all" accordingly.

**Pull request merged**